src/main/java/org/wso2/carbon/inbound/cdc/CDCProcessor.java

```diff
@@ -0,0 +1,239 @@
+            }
+            if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE).equals(FILE_OFFSET_STORAGE_CLASS)) {
+                String filePath;
+                if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME) == null) {
+                    filePath = "cdc/offsetStorage/" + this.name + "_.dat";
+                } else {
+                    filePath = this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME);
+                }
+                createFile(filePath);
+                this.cdcProperties.setProperty(DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME, filePath);
+            }
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_FLUSH_INTERVAL_MS) == null) {
+                this.cdcProperties.setProperty(DEBEZIUM_OFFSET_FLUSH_INTERVAL_MS, "1000");
+            }
+
+            String passwordString = this.cdcProperties.getProperty(DEBEZIUM_DATABASE_PASSWORD);
+            SynapseEnvironment synapseEnvironment = this.synapseEnvironment;
+
+            this.cdcProperties.setProperty(DEBEZIUM_DATABASE_PASSWORD, SecureVaultResolver.resolve(synapseEnvironment, passwordString));
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_DATABASE_ALLOW_PUBLIC_KEY_RETRIEVAL) == null) {
+                this.cdcProperties.setProperty(DEBEZIUM_DATABASE_ALLOW_PUBLIC_KEY_RETRIEVAL, TRUE);
+            }
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_ALLOWED_OPERATIONS) != null) {
+                this.cdcProperties.setProperty(DEBEZIUM_SKIPPED_OPERATIONS,
+                        getSkippedOperationsString(this.cdcProperties.getProperty(DEBEZIUM_ALLOWED_OPERATIONS)));
+            }
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_TOPIC_PREFIX) == null) {
+                this.cdcProperties.setProperty(DEBEZIUM_TOPIC_PREFIX, this.name +"_topic");
+            }
+
+            // set the output format as json in a way a user cannot override
+            this.cdcProperties.setProperty(DEBEZIUM_VALUE_CONVERTER, "org.apache.kafka.connect.json.JsonConverter");
+            this.cdcProperties.setProperty(DEBEZIUM_KEY_CONVERTER, "org.apache.kafka.connect.json.JsonConverter");
+            this.cdcProperties.setProperty(DEBEZIUM_KEY_CONVERTER_SCHEMAS_ENABLE, TRUE);
+            this.cdcProperties.setProperty(DEBEZIUM_VALUE_CONVERTER_SCHEMAS_ENABLE, TRUE);
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_SCHEMA_HISTORY_INTERNAL) == null) {
+                this.cdcProperties.setProperty(DEBEZIUM_SCHEMA_HISTORY_INTERNAL, FILE_SCHEMA_HISTORY_STORAGE_CLASS);
+            }
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_SCHEMA_HISTORY_INTERNAL).equals(FILE_SCHEMA_HISTORY_STORAGE_CLASS)) {
+                String filePath;
+                if (this.cdcProperties.getProperty(DEBEZIUM_SCHEMA_HISTORY_INTERNAL_FILE_FILENAME) == null) {
+                    filePath = "cdc/schemaHistory/" + this.name + "_.dat";
```

Comment:
Shouldn't this be platform-dependent? Or is it handled by the createFile() method?

Reply:
Fixed